core/src/main/scala/kafka/metrics/KafkaMetricsReporter.scala

```diff
@@ -62,7 +63,7 @@ object KafkaMetricsReporter {
         val metricsConfig = new KafkaMetricsConfig(verifiableProps)
         if (metricsConfig.reporters.nonEmpty) {
           metricsConfig.reporters.foreach(reporterType => {
-            val reporter = CoreUtils.createObject[KafkaMetricsReporter](reporterType)
```

Comment:
We could apply the same approach to DumpLogSegments, which would then allow us to remove CoreUtils.createObject.

https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/tools/DumpLogSegments.scala#L649
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/tools/DumpLogSegments.scala#L650

Reply:
I kept the logic in `DumpLogSegments` because once https://github.com/apache/kafka/pull/19030 gets merged we will be able to rewrite `DumpLogSegments`. So I did not want to change the Scala code that will be deleted hopefully very soon (I have a PR almost ready).

I guess while I'm touching `CoreUtils`, I may as well delete this method. I'll push an update.

Reply:
Done